src/uct/ib/ud/base/ud_ep.c

```diff
@@ -977,6 +995,20 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
     }
 
     if (ucs_unlikely(!is_am)) {
+        if ((neth->packet_type & UCT_UD_PACKET_FLAG_FIN) &&
+            (ep->flags & UCT_UD_EP_FLAG_RX)) {
+            uct_ud_iface_cep_remove_ep(iface, ep);
+
+            if (!(ep->flags & UCT_UD_EP_FLAG_TX)) {
+                ep->flags &= ~UCT_UD_EP_FLAG_RX;
```

Comment:
can we do it before if/else? dup line 1006

Reply:
done